### PR TITLE
Work experiences backend

### PR DESCRIPTION
**Pull request by sydneypun**

### CODE REVIEW

File: src/server/src/entities/workExperience.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Entity()
+export class WorkExperience extends BaseEntity {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column()
+    description: string;
+
+    @Column()
+    startDate: string;
```
Comment (johan1505) [inline src/server/src/entities/workExperience.ts]: string -> Date

File: src/server/src/entities/workExperience.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Column({ nullable: true })
+    endDate?: string;
```
Comment (johan1505) [inline src/server/src/entities/workExperience.ts]: string -> Date

File: src/server/src/entities/student.ts [hunk 3]
```diff
@@ -52,6 +53,15 @@ export class Student extends BaseEntity {
     @Column()
     userId: number;
 
+    @OneToOne(() => WorkExperience, {
```
Comment (johan1505) [inline src/server/src/entities/student.ts]: I think there should be a one-to-many relationship between the student and workExperien tables. "A student can have many work experiences but each work experience only belongs to one student"

**Review by johan1505 — CHANGES_REQUESTED** [review]

File: src/server/src/entities/student.ts (hunk 3 above)
Comment (sydneypun) [reply src/server/src/entities/student.ts]: Oh, I see... I'll make the change.

File: src/server/src/entities/workExperience.ts [hunk 4]
```diff
@@ -9,14 +10,17 @@ export class WorkExperience extends BaseEntity {
...
-    endDate?: string;
+    endDate?: Date;
 
     @Column()
     employer: string;
 
     @Column()
     title: string;
+
+    @ManyToOne(() => Student, (student) => student.jobApplications)
```
Comment (johan1505) [inline src/server/src/entities/workExperience.ts]: student.jobApplications  => student.workExperiences

**Review by johan1505 — CHANGES_REQUESTED** [review]
Minor change

**Review by johan1505 — APPROVED** [review]

**Pull request merged**